chia/full_node/weight_proof.py

```diff
@@ -1245,10 +1245,10 @@ def validate_recent_blocks(
         if (challenge is not None) and (prev_challenge is not None):
             overflow = is_overflow_block(constants, block.reward_chain_block.signage_point_index)
             if not adjusted:
+                assert prev_block_record is not None
```

Comment:
At this line, isn't prev_block_record guaranteed to be None, since it was set to None on line 1224?

Reply:
Looks like a couple paths anyways hit an `except Exception:` and others don't, but I didn't find a case where it would catch `TypeError` but not `AssertionError` so I think this is a no-op change give or take.

Reply:
The dataclasses.replace line would raise on None in any event, so this is ok